### PR TITLE
NMS-12279: Cleanup interfaces tagged for flows

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-12279

### CODE REVIEW

File: core/schema/src/main/liquibase/26.0.0/changelog.xml [hunk 1]
```diff
@@ -88,4 +88,138 @@
...
+            <column name="last_egress_flow" type="TIMESTAMP WITH TIME ZONE"  />
+        </addColumn>
+    </changeSet>
+
+    <changeSet author="cpape" id="26.0.0-has-flows-node">
+        <preConditions onFail="MARK_RAN">
+            <columnExists tableName="node" columnName="hasflows" />
+        </preConditions>
+
+        <dropView viewName="node_outages"/>
```
[the remark this answers is not in the document]
Comment (christianpape) [reply core/schema/src/main/liquibase/26.0.0/changelog.xml]: Right

File: features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java [hunk 2]
```diff
@@ -193,11 +196,28 @@ public ElasticFlowRepository(MetricRegistry metricRegistry, JestClient jestClien
         logMarkingTimer = metricRegistry.timer("logMarking");
         flowsPerLog = metricRegistry.histogram("flowsPerLog");
 
-        // Pre-populate marker cache with values from DB
+        this.markerCache.put(Direction.INGRESS, CacheBuilder.newBuilder()
+                .expireAfterWrite(1, TimeUnit.HOURS)
+                .build());
+
+        this.sessionUtils.withTransaction(() -> {
```
[the remark this answers is not in the document]
Comment (christianpape) [reply features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java]: Done

File: core/schema/src/main/liquibase/26.0.0/changelog.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (j-white) [reply core/schema/src/main/liquibase/26.0.0/changelog.xml]: That sucks :(

Can we add a comment to make this clear?

File: opennms-base-assembly/src/main/filtered/etc/opennms.properties [hunk 3]
```diff
@@ -874,3 +874,14 @@ org.opennms.newts.nan_on_counter_wrap=true
 # Defines which endpoints are dispatched to the "osgi rest provider".
 # By default /rest and /api/v2 resources are dispatched.
 # org.opennms.features.osgi.bridge.restAliases=/rest,/api/v2
+
+# ###### Flow Indicator Cleanup ######
+# An interface is marked if flow data has been received within the number of seconds specified here.
+# By default this value is 604800 seconds (= 7 days)
+# org.opennms.features.telemetry.maxFlowAge=604800
```
[the remark this answers is not in the document]
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/opennms.properties]: done

**Comment by j-white** [comment]
Thanks for the updates. Can we re-target this to `release-25.2.1`? :)

File: features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java [hunk 4]
```diff
@@ -167,7 +170,7 @@
      *
      * This maps a node ID to a set if snmpInterface IDs.
      */
-    private final ConcurrentMap<Integer, Set<Integer>> markerCache = Maps.newConcurrentMap();
+    private Map<Direction, Cache<Integer, Set<Integer>>> markerCache = new TreeMap<>();
```
Comment (fooker) [inline features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java]: (Beside the missing `final` ...) a `TreeMap` used for two variables seems to be overkill. There are cases of conditional handling below anyways.
A `Map#get()` and `Map#put()` compared to reading and writing two variable seems to be wrong.
If we decide to keep the `Map`, there is `EnumMap`...

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java [hunk 5]
```diff
@@ -448,14 +449,51 @@ public void setSysName(String nodesysname) {
...
+            return getHasIngressFlows() || getHasEgressFlows();
+        }
+    }
+
+    @Transient
+    public boolean getHasIngressFlows() {
+        if (m_lastIngressFlow == null) {
+            return false;
+        }
+        return (new Date().getTime() - m_lastIngressFlow.getTime()) / 1000 < OnmsSnmpInterface.MAX_FLOW_AGE;
```
Comment (fooker) [inline opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java]: `new Date().getTime()` can be replaced by `System.currentTimeMillis()`

**Comment by fooker** [comment]
I would like to turn around the arrows (ingress should go left, egress should go right).
Or maybe we could add another config toggle for this :see_no_evil:

File: opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java (hunk 5 above)
Comment (christianpape) [reply opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java]: done

**Comment by christianpape** [comment]
> I would like to turn around the arrows (ingress should go left, egress should go right).
> Or maybe we could add another config toggle for this 🙈

done

**Pull request merged**